extensions/guacamole-auth-cas/src/main/java/org/apache/guacamole/auth/cas/conf/PrivateKeyGuacamoleProperty.java

```diff
@@ -49,38 +49,45 @@ public PrivateKey parseValue(String value) throws GuacamoleServerException {
         if (value == null || value.isEmpty())
             return null;
 
+        FileInputStream keyStreamIn = null;
+
         try {
+            try {
```

Comment:
Rather than nest `try` blocks and initialize with `null`, you could always just move the handling for `FileNotFoundException` up:

    FileInputStream keyStreamIn;
    try {
        keyStreamIn = new FileInputStream(value);
    }
    catch (FileNotFoundException e) {
        throw new GuacamoleServerException("Could not find the specified key file.", e);
    }

Since `keyStreamIn` would then be guaranteed to be non-null, complexity of the rest would be reduced, and all you would need is a simple `close()` within a `finally` at the end, like you had before.

Reply:
So, the problem I'm running into is that the `close()` operation can actually throw an `IOException`, which is why I have to nest the `try{}` statements.  Apparently with Java 7+ it gets a little easier, but 6 is a little onerous:

https://stackoverflow.com/questions/156508/closing-a-java-fileinputstream

Reply:
Ah, indeed.